bitnami/tomcat/templates/_pod.tpl

```diff
@@ -77,13 +77,21 @@ containers:
     env:
       - name: BITNAMI_DEBUG
         value: {{ ternary "true" "false" .Values.image.debug | quote }}
+      {{- if and .Values.existingSecret .Values.secretKeys.adminUsernameKey }}
```

Comment:
I would only check `.Values.existingSecret` and set a default value to `tomcat.adminUsernameKey` in the same way as `tomcat.adminPasswordKey`. What do you think?